lib/llvm-backend/src/backend.rs

```diff
@@ -273,25 +287,136 @@ impl LLVMBackend {
+                llvm_backend_get_stack_map_size(module),
+            )
+        };
+        if raw_stackmap.len() > 0 {
+            let map = stackmap::StackMap::parse(raw_stackmap).unwrap();
+
+            let (code_ptr, code_size) = unsafe {
+                (
+                    llvm_backend_get_code_ptr(module),
+                    llvm_backend_get_code_size(module),
+                )
+            };
+            let mut msm = ModuleStateMap {
+                local_functions: Default::default(),
+                total_size: code_size,
+            };
+
+            let mut local_func_id_to_addr: Vec<usize> = Vec::new();
+
+            // All local functions.
+            for index in module_info.imported_functions.len()..module_info.func_assoc.len() {
+                let name = if cfg!(target_os = "macos") {
+                    format!("_fn{}", index)
+                } else {
+                    format!("fn{}", index)
+                };
+
+                let c_str = CString::new(name).unwrap();
+                let ptr = unsafe { get_func_symbol(module, c_str.as_ptr()) };
+
+                assert!(!ptr.is_null());
+                local_func_id_to_addr.push(ptr as usize);
+            }
+
+            let mut addr_to_size_record: BTreeMap<usize, &StkSizeRecord> = BTreeMap::new();
+
+            for record in &map.stk_size_records {
+                addr_to_size_record.insert(record.function_address as usize, record);
+            }
+
+            let mut map_records: BTreeMap<usize, &StkMapRecord> = BTreeMap::new();
+
+            for r in &map.stk_map_records {
+                map_records.insert(r.patchpoint_id as usize, r);
+            }
+
+            for ((start_id, start_entry), (end_id, end_entry)) in stackmaps
+                .entries
+                .iter()
+                .enumerate()
+                .step_by(2)
+                .zip(stackmaps.entries.iter().enumerate().skip(1).step_by(2))
+            {
+                if let Some(map_record) = map_records.get(&start_id) {
+                    assert_eq!(start_id, map_record.patchpoint_id as usize);
+                    assert!(start_entry.is_start);
+                    assert!(!end_entry.is_start);
+
+                    let end_record = map_records.get(&end_id);
+
+                    let addr = local_func_id_to_addr[start_entry.local_function_id];
+                    let size_record = *addr_to_size_record
+                        .get(&addr)
+                        .expect("size_record not found");
+
+                    start_entry.populate_msm(
+                        module_info,
+                        code_ptr as usize,
+                        &map,
+                        size_record,
+                        map_record,
+                        end_record.map(|x| (end_entry, *x)),
+                        &mut msm,
+                    );
+                } else {
+                    // TODO: optimized out?
```

Comment:
Please handle this branch.

Reply:
If the map record is optimized out then doing nothing here is the correct behavior.